server/routes/immediate-needs-report/immediateNeedsReportController.ts

```diff
@@ -26,24 +26,29 @@ export default class ImmediateNeedsReportController {
       const assessmentType = parseAssessmentType(req.query.type)
       const stateKey = {
         prisonerNumber: prisonerData.personalDetails.prisonerNumber,
-        sessionId: req.sessionID,
+        userId: req.user.username,
         pathway,
       }
 
-      // Reset the cache at the point as starting new journey through the form
-      await this.assessmentStateService.reset(stateKey, pathway)
+      const existingInput = await this.assessmentStateService.prepareSubmission(stateKey)
+      const { questionsAndAnswers } = existingInput
+      let currentPageId = null
+
+      if (questionsAndAnswers.length > 0) {
```

Comment:
`if (questionsAndAnswers && questionsAndAnswers.length > 0) {` for safety?

Reply:
the state service has got `?? : {questionsAndAnswers: []}` in it